pkg/registry/core/shoot/storage/admin_kubeconfig.go

```diff
@@ -224,9 +225,32 @@ func findNewestCACertificate(results []*gardencorev1alpha1.GardenerResourceData)
 }
 
 func getClusterCABundle(resourceDataList gardencorev1alpha1helper.GardenerResourceDataList) ([]byte, error) {
-	var caBundle []byte
+	var (
+		allCAs   = resourceDataList.Select(caCertificateSelector.Add(nameCAClusterReq))
+		caBundle []byte
+		err      error
+	)
 
-	for _, data := range resourceDataList.Select(caCertificateSelector.Add(nameCAClusterReq)) {
+	// Sort CAs descending based on their issued-at-time label. This ensures that the current CA is always the first so
+	// that the order in the bundle is the same as in the <shoot-name>.ca-cluster secret.
+	sort.Slice(allCAs, func(i, j int) bool {
+		issuedAtUnix1, err1 := parseIssuedAtUnix(allCAs[i])
+		if err1 != nil {
+			err = err1
+		}
+
+		issuedAtUnix2, err2 := parseIssuedAtUnix(allCAs[j])
+		if err2 != nil {
+			err = err2
+		}
+
+		return issuedAtUnix1 > issuedAtUnix2
+	})
+	if err != nil {
```

Comment:
Does it make sense to add a test case that checks when an error happens?
Also I wonder if there's any way to print all the certificates' names which either do not have `LabelKeyIssuedAtTime ` or the `LabelKeyIssuedAtTime` value could not be parsed.

Reply:
> Does it make sense to add a test case that checks when an error happens?

I have added a test case for this scenario. 

> Also I wonder if there's any way to print all the certificates' names which either do not have LabelKeyIssuedAtTime or the LabelKeyIssuedAtTime value could not be parsed.

Let's not overkill it here now ;-) This (missing `issued-at-time` label) is anyways a highly unlikely scenario since it can only go missing if a human operator drops it manually from the secret in the seed. And if it still happens and a user presents this error message to an operator then they will most likely anyways check all CA secrets in the seed to find out why/what is going on.